lib/simpleanidb/models.py

```diff
@@ -0,0 +1,230 @@
+from __future__ import absolute_import
+import requests
+import xml.etree.ElementTree as ET
+
+from .helper import date_to_date
+from .exceptions import GeneralError
+
+
+class Anime(object):  # pylint: disable=too-many-instance-attributes
+    def __init__(self, anidb, aid, auto_load=True, xml=None):
+        self.anidb = anidb
+        self.aid = aid
+        self.titles = []
+        self.synonyms = []
+        self.all_episodes = []
+        self.episodes = {}
+        self.picture = None
+        self.rating_permanent = None
+        self.rating_temporary = None
+        self.rating_review = None
+        self.categories = []
+        self.tags = []
+        self.start_date = None
+        self.end_date = None
+        self.description = None
+
+        if xml:
+            self.fill_from_xml(xml)
+
+        self._loaded = False
+        if auto_load:
+            self.load()
+
+    def __repr__(self):
+        return "<Anime:{0} loaded:{1}>".format(self.aid, self.loaded)
+
+    @property
+    def loaded(self):
+        return self._loaded
+
+    def load(self):
+        """Load all extra information for this anime.
+
+        The anidb url should look like this:
+        http://api.anidb.net:9001/httpapi?request=anime&client={str}&clientver={int}&protover=1&aid={int}
+        """
+        params = {
+            "request": "anime",
+            "client": "adbahttp",
+            "clientver": 100,
+            "protover": 1,
+            "aid": self.aid
+        }
+
+        self._xml = self.anidb._get_url("http://api.anidb.net:9001/httpapi", params=params)
+
+        self.fill_from_xml(self._xml)
+        self._loaded = True
+
+    def fill_from_xml(self, xml):  # pylint: disable=too-many-branches
+        if xml.find("titles") is not None:
+            self.titles = [Title(self, n) for n in xml.find("titles")]
+        else:
+            self.titles = [Title(self, n) for n in xml.findall("title")]
+            # return # returning from here will result in not loading attribute information for anime lists like hot_animes
+        self.synonyms = [t for t in self.titles if t.type == "synonym"]
+        if xml.find("episodes") is not None:
+            self.all_episodes = sorted([Episode(self, n) for n in xml.find("episodes")])
+            self.episodes = {e.number: e for e in self.all_episodes if e.type == 1}
+        if xml.find("picture") is not None:
+            self.picture = Picture(self, xml.find("picture"))
+        if xml.find("ratings") is not None:
+            if xml.find("ratings").find("permanent") is not None:
+                self.rating_permanent = xml.find("ratings").find("permanent").text
+            if xml.find("ratings").find("temporary") is not None:
+                self.rating_temporary = xml.find("ratings").find("temporary").text
+            if xml.find("ratings").find("review") is not None:
+                self.rating_review = xml.find("ratings").find("review").text
```

Comment:
Hmm, number of votes is in xml, but not parsed. As we use those on the recommended lists, maybe i'll need to add those.